crates/bevy_gizmos/src/pipeline_3d.rs

```diff
@@ -0,0 +1,164 @@
+use bevy_asset::Handle;
+use bevy_core_pipeline::core_3d::Opaque3d;
+use bevy_ecs::{
+    entity::Entity,
+    query::With,
+    system::{Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_pbr::*;
+use bevy_render::{
+    mesh::Mesh,
+    render_resource::Shader,
+    view::{ExtractedView, ViewTarget},
+};
+use bevy_render::{
+    mesh::MeshVertexBufferLayout,
+    render_asset::RenderAssets,
+    render_phase::{DrawFunctions, RenderPhase, SetItemPipeline},
+    render_resource::*,
+    texture::BevyDefault,
+    view::Msaa,
+};
+
+use crate::{GizmoConfig, GizmoMesh, LINE_SHADER_HANDLE};
+
+#[derive(Resource)]
+pub(crate) struct GizmoPipeline {
+    mesh_pipeline: MeshPipeline,
+    shader: Handle<Shader>,
+}
+
+impl FromWorld for GizmoPipeline {
+    fn from_world(render_world: &mut World) -> Self {
+        GizmoPipeline {
+            mesh_pipeline: render_world.resource::<MeshPipeline>().clone(),
+            shader: LINE_SHADER_HANDLE.typed(),
+        }
+    }
+}
+
+impl SpecializedMeshPipeline for GizmoPipeline {
+    type Key = (bool, MeshPipelineKey);
+
+    fn specialize(
+        &self,
+        (depth_test, key): Self::Key,
+        layout: &MeshVertexBufferLayout,
+    ) -> Result<RenderPipelineDescriptor, SpecializedMeshPipelineError> {
+        let mut shader_defs = Vec::new();
+        shader_defs.push("GIZMO_LINES_3D".into());
+        shader_defs.push(ShaderDefVal::Int(
```

Comment:
Not something to fix here, but the fact that need to manually plug these values in, despite not using them in the shader, is unfortunate.